### DISCUSSION TITLE
[3.10] [4.0] Instructions in Joomla Docs for Migration don't work

### DISCUSSION THREAD

**Discussion by Scrabble96** [joomla/joomla-cms#31637] [Q&A]
### Steps to reproduce the issue

Follow the instructions here: https://docs.joomla.org/Joomla_3.x_to_4.x_Step_by_Step_Migration

### Expected result
In J3.10 on development site, select Components -> Joomla Update -> and -> Joomla Next then Save & Close.

Expect to see Pre-Update Check.

### Actual result

Pre-Update Check not shown:

![Joomla Next](https://user-images.githubusercontent.com/36987494/101918700-92a5d300-3bc1-11eb-9b49-39586705e187.jpg)

### System information (as much as possible)

Web sub-domain (not local)
PHP 7.4

**Comment by brianteeman** [comment]
That is because, as the text sys, that update channel will only show you when yhe release is available.

**Comment by Scrabble96** [comment]
UPDATE: If you follow the instructions here: https://www.joomla.org/announcements/release-news/5822-joomla-4-0-0-beta4-and-joomla-3-10-alpha2.html it tells you to switch from 'Next' to 'Custom URL' and then you see the Pre-Update Check.

**Comment by Scrabble96** [comment]
Further update: the upgrade doesn't seem to like it if either (or possibly both) 
a) your J3 website is not currently using Protostar and/or
b) Protostar is disabled

**Comment by wilsonge** [comment]
> Further update: the upgrade doesn't seem to like it if either (or possibly both)
> a) your J3 website is not currently using Protostar and/or
> b) Protostar is disabled

Can you elaborate or give some extra information? This shouldn't really affect things

**Comment by Scrabble96** [comment]
> 
> 
> > Further update: the upgrade doesn't seem to like it if either (or possibly both)
> > a) your J3 website is not currently using Protostar and/or
> > b) Protostar is disabled
> 
> Can you elaborate or give some extra information? This shouldn't really affect things

Well, I've just tried it again on two sites using two different custom templates and it now works. I don't know what was happening before (except a lot of wasted time). It was very odd. I was getting error messages such as 'Class 'JResponse' not found', and '500 Unknown column fp.featured-up' in 'field list''. But not today.

Going back to my original post, the instructions have been written as if J4 is fully released. There should be a note added to the top explaining that for testing purposes with J3.10 alpha/beta and J4.0 alpha/beta before the main release that the method for upgrading to J4 Beta x is slightly different. For example, having upgraded from J3.9x to J3.10.0.0 alpha3 to get from there to the J4 Beta x version the 'Custom URL' and 'Beta' options should be selected in the Joomla update options and this URL - https://update.joomla.org/core/test/310to4_list.xml - should be used.

**Comment by rdeutz** [comment]
@zero24 can you have a look here, thanks

**Comment by zero-24** [comment]
As mention above that article is using the live upgrade instruction. I'm fine with that there can be a note for pre final releases as mention by @Scrabble96
